tck/src/main/java/ee/jakarta/tck/data/core/cdi/ExtensionTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.data.core.cdi;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import ee.jakarta.tck.data.core.cdi.provider.PersonExtension;
+import ee.jakarta.tck.data.framework.junit.anno.AnyEntity;
+import ee.jakarta.tck.data.framework.junit.anno.Assertion;
+import ee.jakarta.tck.data.framework.junit.anno.CDI;
+import ee.jakarta.tck.data.framework.junit.anno.Core;
+import jakarta.enterprise.inject.spi.Extension;
+import jakarta.inject.Inject;
+
+@Core
+@AnyEntity
+@CDI
```

Comment:
This test is only valid in web profile (or platform), where CDI Full is available, providing `jakarta.enterprise.inject.spi.Extension` which is used for the extension.
In core profile, only CDI Lite is required, which is not required to include `jakarta.enterprise.inject.spi.Extension`.
Note that `jakarta.enterprise.inject.build.compatible.spi.BuildCompatibleExtension` must be available in both CDI Lite and CDI Full, meaning it can be relied upon for all of Core Profile, Web Profile, and Jakarta EE Platform.